Objects/sliceobject.c

```diff
@@ -628,6 +628,42 @@ slice_traverse(PySliceObject *v, visitproc visit, void *arg)
     return 0;
 }
 
+/* code based on tuplehash() of Objects/tupleobject.c */
```

Comment:
This is all nice work; however, a simpler solution will suffice.

```
def slice_hash(s):
    # Multiply by two randomly chosen primes
    return hash(s.start) * 314898481 | hash(s.stop) * 4822421 | hash(s.step)
```

Reply:
After more thought, I think we can leave your version as-is.  If someone wants to propose a simplification and test collision rates, they can do so at a later time.  It is reasonable to stick with the known solution.